widgetastic_manageiq/__init__.py

```diff
@@ -2635,6 +2637,127 @@ def fill(self, items):
+            self.locator = locator
+        else:
+            raise TypeError("You need to specify a locator for ReactSelect")
+
+        self.can_hide_on_select = can_hide_on_select
+
+    @property
+    def is_open(self):
+        try:
+            return self.browser.is_displayed('//div[@class="css-1paxw40-menu"]')
+        except StaleElementReferenceException:
+            self.logger.warning(
+                "Got a StaleElementReferenceException in .is_open, but ignoring. Returned False."
+            )
+            return False
+
+    def open(self):
+        if not self.is_open:
+            self.click()
+            self.logger.debug("opened")
+
+    def close(self):
+        try:
+            if self.is_open:
+                self.click()
+                self.logger.debug("closed")
+        except NoSuchElementException:
+            if self.can_hide_on_select:
+                self.logger.info("While closing %r it disappeared, but ignoring.", self)
+            else:
+                raise
+
+    def select_by_visible_text(self, *items):
+        """Selects items in the select.
+
+        Args:
+            *items: Items to be selected. If the select does not support multiple selections and you
+                pass more than one item, it will raise an exception. If you want to select using
+                partial match, use the :py:class:`BootstrapSelect.partial` to wrap the value.
+        """
+        if len(items) > 1:
+            raise ValueError(
+                "The ReactSelect {} does not allow multiple selections".format(self.locator)
+            )
+        self.open()
+        for item in items:
+            item = item.split(" *")[0]
+            self.logger.info("selecting by visible text: %r", item)
+            try:
+                self.browser.click(
+                    self.BY_VISIBLE_TEXT.format(quote(item)), parent=self, force_scroll=True
+                )
+            except NoSuchElementException:
+                try:
+                    # Added this as for some views(some tags pages) dropdown is separated from
+                    # button and doesn't have exact id or name
+                    self.browser.click(self.BY_VISIBLE_TEXT.format(quote(item)), force_scroll=True)
+                except NoSuchElementException:
+                    raise SelectItemNotFound(
+                        widget=self, item=item, options=[opt.text for opt in self.all_options]
+                    )
+        self.close()
+
+    @property
+    def all_selected_options(self):
+        return [
+            self.browser.text(e)
+            for e in self.browser.elements('//ul[@class="tag-category list-inline"]', parent=self)
+        ]
+
+    @property
+    def all_options(self):
+        script = """return ManageIQ.component.getInstance(
+        "TaggingWrapperConnected", "TaggingWrapperConnected-0").props.tags"""
+        return self.browser.execute_script(script)
+
+    @property
+    def selected_option(self):
```

Comment:
This returns an `IndexError` for me when only the `tag_category` is selected (i.e. no `tag_value` selected). 
```python,
view.form.tag_category.selected_option
Traceback (most recent call last):
  File "/home/jdupuy/cfme_tests/cfme_venv/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-13-5a608d12944b>", line 1, in <module>
    view.form.tag_category.selected_option
  File "/home/jdupuy/cfme_tests/integration_tests/widgetastic_manageiq/__init__.py", line 2740, in selected_option
    return self.all_selected_options[0]
IndexError: list index out of range
```

Additionally, when the `tag_value` is also selected, it shows both the `tag_category` and the `tag_value`.

Reply:
thanks John! I misread this function and implemented something different, let me try to do it correctly

Reply:
fixed